src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSchemas.cs

```diff
@@ -16,40 +17,41 @@ namespace System.Xml.Serialization
     using System.Security;
     using System.Net;
     using System.Reflection;
+    using System.Diagnostics.CodeAnalysis;
 
     public class XmlSchemas : CollectionBase, IEnumerable<XmlSchema>
     {
-        private XmlSchemaSet _schemaSet;
-        private Hashtable _references;
-        private SchemaObjectCache _cache; // cached schema top-level items
+        private XmlSchemaSet? _schemaSet;
+        private Hashtable? _references;
+        private SchemaObjectCache? _cache; // cached schema top-level items
         private bool _shareTypes;
-        private Hashtable _mergedSchemas;
+        private Hashtable? _mergedSchemas;
         internal Hashtable delayedSchemas = new Hashtable();
         private bool _isCompiled;
-        private static volatile XmlSchema s_xsd;
-        private static volatile XmlSchema s_xml;
+        private static volatile XmlSchema? s_xsd;
+        private static volatile XmlSchema? s_xml;
 
         public XmlSchema this[int index]
         {
-            get { return (XmlSchema)List[index]; }
+            get { return (XmlSchema)List[index]!; }
```

Comment:
If you're going to make the return type nullable, you'd also have to remove the `!` here I think

Reply:
See my comment https://github.com/dotnet/runtime/pull/41261#discussion_r477302630